gfx.cpp

```diff
@@ -2162,7 +2230,26 @@ class GfxInternal
         RaytracingPrimitive const &gfx_raytracing_primitive = raytracing_primitives_[raytracing_primitive];
         GfxAccelerationStructure const &acceleration_structure = getRaytracingPrimitiveAccelerationStructure(gfx_raytracing_primitive);
         if(acceleration_structure_handles_.has_handle(acceleration_structure.handle))
+        {
             acceleration_structures_[acceleration_structure].needs_rebuild_ = true;
+            if(gfx_raytracing_primitive.type_ == RaytracingPrimitive::kType_Procedural)
+            {
+                auto &primitives = acceleration_structures_[acceleration_structure].raytracing_primitives_;
+                if(gfx_raytracing_primitive.index_ < primitives.size() && primitives[gfx_raytracing_primitive.index_].handle == raytracing_primitive.handle)
```

Comment:
I noticed that a primitive is not being removed from the list of all primitives in AS, so this fixes the issue.
Previously, it wasn't a problem since the object were loaded and deleted at once during scene init. But with introduction of AABBs, that's not always that case.

I'm not sure if it's 100% correct to shift indices to the left so I put an `if` so that this code doesn't trigger in usual use cases to avoid other potential issues. What do you think about it? @gboisse @maoliver-amd 
```
--primitive_ahead.index_;
--primitive_ahead.instance_id_;
```